pkg/controller/core/workload_controller.go

```diff
@@ -242,6 +357,14 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 		}
 	}
 
+	if status == admitted && r.podsReadyTimeout != nil {
+		if key, err := toolscache.MetaNamespaceKeyFunc(wl); err != nil {
+			log.Error(err, "Failed to create a workload key")
+		} else {
+			r.podsReadyTimeoutSyncQueue.Add(key)
+		}
+	}
```

Comment:
I'm not convinced that we needed a separate sync instead of adding the logic the existing `Reconcile` method. `ctrl.Result` has the necessary options to delay the next reconcile.

A separate sync would be recommended if we need to deal with different objects, like the Job controller leading with Jobs and Pods, or the disruption controller dealing with PDBs and Pods. This controller is just dealing with Workloads.

Reply:
It might not be supported. You can ask in slack #controller-runtime and open an issue if there is no support for it.

Reply:
Asked https://kubernetes.slack.com/archives/C02MRBMN00Z/p1672929287689979